### PR TITLE
Freshen up `Middleware` docs

### LINKED ISSUE #5086: middleware.md still refers to HttpService
Thanks to @nafg for pointing that out [here](https://github.com/http4s/http4s/pull/1693#issuecomment-900737068).

### PR DESCRIPTION
**Pull request by danicheg**
Fixes #5086.
I did it very straightforwardly. If you have something to add, let me know.

### CODE REVIEW

File: docs/docs/middleware.md [hunk 1]
```diff
@@ -1,14 +1,13 @@
 # Middleware
 
-A middleware is a wrapper around a [service] that provides a means of manipulating
```
Comment (danicheg) [inline docs/docs/middleware.md:3]: I can be biased, but explaining things using 'wrapper' things doesn't help much.

File: docs/docs/middleware.md [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+A middleware is an abstraction around a [service] that provides a means of manipulating
 the `Request` sent to service, and/or the `Response` returned by the service. In
 some cases, such as [Authentication], middleware may even prevent the service
 from being called.
 
-At its most basic, middleware is simply a function that takes one `Service` as a
-parameter and returns another `Service`. In addition to the `Service`, the middleware
-function can take any additional parameters it needs to perform its task. Let's look
-at a simple example.
+At its most basic, middleware is simply a function that takes one service
```
Comment (ChristopherDavenport) [inline docs/docs/middleware.md:8]: The word simply here has always bugged me.

File: docs/docs/middleware.md [hunk 3]
```diff
@@ -107,29 +108,35 @@ newService.orNotFound(goodRequest).unsafeRunSync()
...
 ```
 
-It is possible for the wrapped `Service` to have different `Request` and `Response`
-types than the middleware. Authentication is, again, a good example. Authentication
-middleware is an `HttpService` (an alias for `Service[Request, Response]`) that wraps an `
-AuthedService` (an alias for `Service[AuthedRequest[T], Response]`. There is a type
-defined for this in the `http4s.server` package:
+Let's consider Authentication middleware as an example. Authentication
+middleware is a function that takes `AuthedRoutes[F]` 
+(an alias for `Kleisli[OptionT[F, *], AuthedRequest[F, T], Response[F]]`)
```
Comment (ChristopherDavenport) [inline docs/docs/middleware.md:113]: Instead of kleisli what about explaining it in terms of `AuthedRequest[F, T] => F[Option[Response[F]]]`

Comment (danicheg) [reply docs/docs/middleware.md:113]: good call, will fix this

File: docs/docs/middleware.md [hunk 4]
```diff
@@ -107,29 +108,35 @@ newService.orNotFound(goodRequest).unsafeRunSync()
...
 
-It is possible for the wrapped `Service` to have different `Request` and `Response`
-types than the middleware. Authentication is, again, a good example. Authentication
-middleware is an `HttpService` (an alias for `Service[Request, Response]`) that wraps an `
-AuthedService` (an alias for `Service[AuthedRequest[T], Response]`. There is a type
-defined for this in the `http4s.server` package:
+Let's consider Authentication middleware as an example. Authentication
+middleware is a function that takes `AuthedRoutes[F]` 
+(that translates to `AuthedRequest[F, T] => F[Option[Response[F]]]`) 
+and returns `HttpRoutes[F]` (that translates to `Request[F, T] => F[Option[Response[F]]]`).
```
Comment (ChristopherDavenport) [inline docs/docs/middleware.md:114]:
```suggestion
and returns `HttpRoutes[F]` (that translates to `Request[F] => F[Option[Response[F]]]`). 
```

**Review by ChristopherDavenport — COMMENTED** [review]
I love it, theres a fix thats still needed but I like it a lot.

**Comment by ChristopherDavenport** [comment]
My wonder I've always had is what if we generally omitted the `Request[F]` to just call in `Request` docs. This is more a discussion point than something I want to see addressed here. That way when explaining it would look like `Request => F[Option[Response]]` which hides the noise that has to do with the internal stream bodies.

File: docs/docs/middleware.md (hunk 4 above)
Comment (danicheg) [reply docs/docs/middleware.md:114]: 🤦🏻  thank you for pointing this out

**Comment by danicheg** [comment]
I think that hiding the `F` type parameter is absolutely legit in docs, especially when we're explaining things using pseudo-code. And vice versa, every code snippet should be compiled and ready for copy-pasting by users.

**Review by ChristopherDavenport — APPROVED** [review]

**Pull request merged**